src/vue-cookie.js

```diff
@@ -17,12 +17,31 @@
             if(Number.isInteger(daysOrOptions)) {
                 opts = {expires: daysOrOptions};
             }
+
             return Cookie.set(name, value, opts);
         },
+        setObject: function(name, value, daysOrOptions) {
+            try {
+                var result = JSON.stringify(value);
+                if (/^[\{\[]/.test(result)) {
+                    value = result;
+                }
+            } catch (e) {}
 
+            return this.set(name, value, opts);
```

Comment:
I see only one problem, here you are passing the third parameter ```opts``` to ```this.set()``` but the third parameter for ```setObject()``` is called ```daysOrOptions``` thus you will always be passing ```undefined``` to ```this.set()```